### PR TITLE
Issue 372

### PR DESCRIPTION
**Pull request by ssiegel95**
* Adds support for reading multi part CSV files
* Assumptions:
   - all csv files have the same header row in the first line
   - user specified as path url such as `file:///tmp/multipart` and that the `multipart` directory contain **no subdirectories** or files **other than csv files** which are identified simply by a `csv` extension.
   - if any of the assumptions are broken the user will get a ValueError exception back

None of this enables iterative datasets (that is planned for the future once more iterative dataset handling is introduced in the base tsfm_public module. Thus for now, this will still lead to a full in-memory load of the data into a pandas dataframe.

* We will have to test for edge cases around subtle differences in how pyarrow parses csv content compared to pandas. The main difference I've seen so far is in timestamps (what else). The former seems to do a better job at detecting the necessary level of precision, using second level precision where pandas seems to unnecessarily use ns level precision.

This PR is intended to address #372

### CODE REVIEW

File: services/finetuning/tsfmfinetuning/ioutils.py [hunk 1]
```diff
@@ -92,7 +151,9 @@ def to_pandas(uri: str, **kwargs) -> pd.DataFrame:
     Returns:
         pd.DataFrame: a pandas DataFrame object
     """
-
+    # print(f"grr data {list(os.walk('data'))}")
```
Comment (wgifford) [inline services/finetuning/tsfmfinetuning/ioutils.py:154]: maybe drop these?

File: services/finetuning/tsfmfinetuning/ioutils.py [hunk 2]
```diff
@@ -47,17 +39,84 @@ def get_object_name(s3_uri):
...
 
 
-def _to_pandas(iscsv: bool, isfeather: bool, buf: BytesIO = None, timestamp_column: str = None):
+def _to_pandas(
+    iscsv: bool, isfeather: bool, buf: Union[str, BytesIO, None] = None, timestamp_column: Union[str, None] = None
+):
     if iscsv:
-        return pd.read_csv(buf, parse_dates=[timestamp_column] if timestamp_column else False)
+        if os.path.isdir(buf):
+            import pyarrow.dataset as ds
```
Comment (wgifford) [inline services/finetuning/tsfmfinetuning/ioutils.py:47]: What are the assumptions of the multipart file? Is any ordering assumed? Is it assumed to be split by different IDs?

**Review by wgifford — COMMENTED** [review]
Tests look comprehensive. Just a few minor questions.

File: services/finetuning/tsfmfinetuning/ioutils.py (hunk 1 above)
Comment (ssiegel95) [reply services/finetuning/tsfmfinetuning/ioutils.py]: done

File: services/finetuning/tsfmfinetuning/ioutils.py (hunk 2 above)
Comment (ssiegel95) [reply services/finetuning/tsfmfinetuning/ioutils.py:47]: There are no assumptions beyond what pyarrow's dataset creation imposes. Pyarrow reads files in ascending lexicographical order and will recurs into all sub-directories (but I restrict the parameter to point to only a flat directory with no sub-directories).  So no assumptions about data ordering or splitting by ID. I was assuming that the preprocessor handles this as it would any other pandas dataframe it gets from a single file.

**Review by wgifford — APPROVED** [review]
